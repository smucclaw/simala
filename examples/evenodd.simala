-- The unfortunately quite complicated way in which we have to
-- encode mutually recursive definitions right now. It would be
-- better if recursive declarations would work for non-functions.
--
let rec evenodd = fun (unused) =>
  { even = fun (x) => x == 0 || evenodd(unused).odd(x - 1)
  , odd  = fun (x) => x == 1 || evenodd(unused).even(x - 1)
  }
in evenodd('irrelevant).even(20)
