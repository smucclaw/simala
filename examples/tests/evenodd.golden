/home/andres/repos/well-typed/cclaw/simala/./examples/evenodd.simala:5:1:
  |
5 | let rec evenodd = fun (unused) =>
  | ^
unexpected 'l'
expecting end of input
